File: scala/sum-of-multiples/src/main/scala/SumOfMultiples.scala
object SumOfMultiples {
  def sum(factors: Set[Int], limit: Int): Int = {
    def isMultiple(number: Int) = factors.exists(factor => number % factor == 0)

    (1 until limit).foldLeft(0)((acc, num) => if (isMultiple(num)) acc + num else acc)
  }
}

File: typescript/sum-of-multiples/sum-of-multiples.ts
export default function SumOfMultiples(multiples: number[]) {
    return new SumOfMultiplesImpl(multiples)
}

class SumOfMultiplesImpl {
    protected multiples: number[]
    constructor(multiples: number[]) {
        this.multiples = multiples
    }

    to(n: number): number {
        let result = 0
        for (let i = 1; i < n; i++) {
            if (this.isMultiple(i)) {
                result = result + i
            }
        }

        return result
    }

    isMultiple(n: number): boolean {
        return !(this.multiples.find((multiple) => n % multiple === 0) === undefined)
    }
}